mlir/lib/Dialect/Affine/Utils/LoopUtils.cpp

```diff
@@ -117,7 +117,8 @@ static void replaceIterArgsAndYieldResults(AffineForOp forOp) {
 /// was known to have a single iteration.
 LogicalResult mlir::affine::promoteIfSingleIteration(AffineForOp forOp) {
   std::optional<uint64_t> tripCount = getConstantTripCount(forOp);
-  if (!tripCount || *tripCount != 1)
+  std::optional<uint64_t> maxTripCount = getUpperBoundOnTripCount(forOp);
+  if (!tripCount || *tripCount != 1 || !maxTripCount || *maxTripCount != 1)
     return failure();
```

Comment:
I'm going to suggest renaming `tripCount` to `minTripCount` to make the logic clearer

Reply:
Sorry, I have some other questions for you, not about this PR. Can you give me some contact information? (Or can I contact you directly through the email address on your github homepage?

Reply:
Yeah, the emails on my GitHub work